### PR TITLE
deal making FRC added

### PR DESCRIPTION
**Pull request by aashidham**

### CODE REVIEW

**Comment by luckyparadise** [comment]
@aashidham would you like to present this FRC at our next Core Devs Call? Might be a good place to gather initial feedback and comments.

**Comment by aashidham** [comment]
> @aashidham would you like to present this FRC at our next Core Devs Call? Might be a good place to gather initial feedback and comments. 

Absolutely! I'll follow up on Slack.

**Comment by aashidham** [comment]
Following up on this -- what is pending before approval can be granted? Started a discussion for this PR to solicit more feedback: https://github.com/filecoin-project/FIPs/discussions/721

**Comment by luckyparadise** [comment]
Hi @aashidham there is no "approval" as such as all FRCs at the moment remain as open drafts. However, if you like, I can share this widely and solicit feedback and engagement for you. Let me know thanks.

File: FRCs/frc-nnnn-deal-making.md [hunk 1]
```diff
@@ -0,0 +1,386 @@
+| fip | title | author | discussions-to | status | type | created |
+| --- | --- | --- | --- | --- | --- | --- |
+| 0060 | Deal making protocol between SPs and FVM smart contracts | Aadi S (@aashidham), Raul K (@raulk), Scott C (@skottie86), Jennifer W (@jennijuju), Anton E, Shrenuj B | https://github.com/filecoin-project/FIPs/discussions/492 | Draft | FRC | 2023-04-25 |
```
Comment (anorth) [inline FRCs/frc-nnnn-deal-making.md:3]: This discussion link doesn't go to the right place, should be https://github.com/filecoin-project/FIPs/discussions/721

File: FRCs/frc-nnnn-deal-making.md [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+
+# Deal making protocol between SPs and FVM smart contracts
+
+# Simple Summary
```
Comment (anorth) [inline FRCs/frc-nnnn-deal-making.md:7]: Please follow the template format, which uses `##` for these headings.

File: FRCs/frc-nnnn-deal-making.md [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+4. Composability: A standard interface for programmatic deal making (proposals, verification, payments) allows both clients, storage providers, and developers to build together efficiency.
+
+# Specification
+
+A valid client contract implementation would minimally support either the `OneToOneDealProposer` or `WildcardDealProposer` interface. Both of these inherit the `BaseDealProposer` interface.
+
+In particular, the `OneToOneDealProposer` interface is for client contracts that have specified a single Boost SP for every deal proposal. Developers implementing this interface will likely have to reach out and coordinate with Boost SPs specified for a particular deal. Boost SPs with address `x` can then listen to `OneToOneDealProposalCreate` events with the provider field set to `x`. This flow is tremendously simplified, since it only deals with one client contract and one Boost SP per deal proposal.
```
Comment (anorth) [inline FRCs/frc-nnnn-deal-making.md:30]: I don't think reference to "Boost" belongs in this FRC. Boost is just one implementation of deal-making.

File: FRCs/frc-nnnn-deal-making.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (aashidham) [reply FRCs/frc-nnnn-deal-making.md]: have a look now?

File: FRCs/frc-nnnn-deal-making.md (hunk 3 above)
Comment (aashidham) [reply FRCs/frc-nnnn-deal-making.md]: I largely agree -- however there are some references to Boost that are useful for context. Have a look at the latest draft.

File: FRCs/frc-nnnn-deal-making.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (aashidham) [reply FRCs/frc-nnnn-deal-making.md]: (feel free to unresolve if you have more template notes)

**Review by anorth — APPROVED** [review]
Editorial 👍

**Review by kaitlin-beegle — APPROVED** [review]

**Review by kaitlin-beegle — APPROVED** [review]

File: FRCs/frc-0067.md [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+    uint64 piece_size;
+    bool verified_deal;
+    string label;
+    int64 start_epoch;
+    int64 end_epoch;
+    uint256 storage_price_per_epoch;
+    uint256 provider_collateral;
+    uint256 client_collateral;
+    uint64 extra_params_version;
+    ExtraParamsV1 extra_params;
```
Comment (nonsense) [inline FRCs/frc-0067.md:233]: I think the idea here is to have the `version` of the `params` (currently we have only one version, namely `ExtraParamsV1`) in the `extra_params_version` field, and for the `extra_params` to be generic, such as `bytes`, and not of specific type as it is defined now.

The way it is defined now make the parameter `extra_params_version` redundant, no?

File: FRCs/frc-0067.md [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
```
Comment (nonsense) [inline FRCs/frc-0067.md:248]: Same comment here - https://github.com/filecoin-project/FIPs/pull/693/files#r1277406267

**Comment by luckyparadise** [comment]
@kaitlin-beegle can you merge this PR and assign the FRC number?

**Comment by kaitlin-beegle** [comment]
> @kaitlin-beegle can you merge this PR and assign the FRC number?

@luckyparadise yes, but please follow up @aashidham next week and make sure they are able to follow up @nonsense's feedback.

**Pull request merged**